### PR TITLE
labeler: add CI-iot-samples-test label

### PR DESCRIPTION
**Pull request by richardmccrae-nordicsemi**
Add Github label and associated files to trigger automated testing for nrf9160 samples.

Signed-off-by: Richard McCrae <richard.mccrae@nordicsemi.no>

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -14,10 +14,26 @@
...
+  - "include/modem/*"
+  - "include/net/coap_utils.h"
+  - "include/net/download_client.h"
+  - "include/net/multicell_location.h"
+  - "include/secure_services.h"
+  - "lib/at_cmd/*"
+  - "lib/at_cmd_parser/*"
+  - "lib/at_host/*"
+  - "lib/at_monitor/*"
+  - "lib/at_notify/*"
```
Comment (jorgenmk) [inline .github/labeler.yml:28]: This should probably be `at_notif`.

File: .github/labeler.yml [hunk 2]
```diff
@@ -14,10 +14,26 @@
   - "doc/**/*"
   - "**/*.rst"
 
-"CI-libmodem-test":
+"CI-iot-samples-test":
```
Comment (jorgenmk) [inline .github/labeler.yml:17]: Perhaps all the folders here could end with `/**/*` in case they later decide to add subfolders?

**Review by jorgenmk — CHANGES_REQUESTED** [review]

**Review by jorgenmk — APPROVED** [review]

File: .github/labeler.yml [hunk 3]
```diff
@@ -14,10 +14,26 @@
...
+  - "lib/at_notif/**/*"
   - "lib/lte_link_control/**/*"
-  - "lib/nrf_modem_lib/*"
+  - "lib_modem_key_mgmt/**/*"
+  - "lib/multicell_location/**/*"
+  - "lib/nrf_modem_lib/**/*"
+  - "lib/pdn/**/*"
+  - "lib/sms/**/*"
+  - "subsys/net/lib/coap_utils/**/*"
+  - "subsys/net/lib/download_client/**/*"
```
Comment (Eoin-McMahon) [inline .github/labeler.yml:36]: for the lwm2m samples we should also add lib/bin/lwm2m_carrier to this list

**Review by Eoin-McMahon — CHANGES_REQUESTED** [review]

File: .github/labeler.yml [hunk 4]
```diff
@@ -14,10 +14,26 @@
...
+  - "include/net/multicell_location.h"
+  - "include/secure_services.h"
+  - "lib/at_cmd/**/*"
+  - "lib/at_cmd_parser/**/*"
+  - "lib/at_host/**/*"
+  - "lib/at_monitor/**/*"
+  - "lib/at_notif/**/*"
   - "lib/lte_link_control/**/*"
-  - "lib/nrf_modem_lib/*"
+  - "lib_modem_key_mgmt/**/*"
```
Comment (richardmccrae-nordicsemi) [inline .github/labeler.yml:30]: This needs to be checked.  probably lib/modem_key_mgmt

Comment (richardmccrae-nordicsemi) [reply .github/labeler.yml]: fixed

File: .github/labeler.yml (hunk 3 above)
Comment (richardmccrae-nordicsemi) [reply .github/labeler.yml:37]: fixed

**Review by Eoin-McMahon — APPROVED** [review]

File: .github/labeler.yml [hunk 5]
```diff
@@ -14,16 +14,33 @@
   - "doc/**/*"
   - "**/*.rst"
 
-"CI-libmodem-test":
+"CI-iot-samples-test":
   - "samples/nrf9160/**/*"
+  - "include/modem/**/*"
+  - "include/net/coap_utils.h"
```
Comment (lemrey) [inline .github/labeler.yml:20]: I do not think this is needed, it is used by Thread.

File: .github/labeler.yml [hunk 6]
```diff
@@ -14,16 +14,33 @@
...
+  - "lib/at_notif/**/*"
+  - "lib/bin/lwm2m_carrier/**/*"
   - "lib/lte_link_control/**/*"
-  - "lib/nrf_modem_lib/*"
+  - "lib/modem_key_mgmt/**/*"
+  - "lib/multicell_location/**/*"
+  - "lib/nrf_modem_lib/**/*"
+  - "lib/pdn/**/*"
+  - "lib/sms/**/*"
+  - "subsys/net/lib/coap_utils/**/*"
```
Comment (lemrey) [inline .github/labeler.yml:36]: This can be removed

File: .github/labeler.yml [hunk 7]
```diff
@@ -14,16 +14,56 @@
   - "doc/**/*"
   - "**/*.rst"
 
-"CI-libmodem-test":
-  - "samples/nrf9160/**/*"
+"CI-iot-samples-test":
+  - "include/modem/at_cmd*.h"
+  - "include/modem/at_monitor.h"
+  - "include/modem/at_notify.h"
```
Comment (lemrey) [inline .github/labeler.yml:20]: This is incorrect, it should be `at_notif.h`

**Review by lemrey — COMMENTED** [review]
Not sure about the changelog entry, @rlubos do we usually track this kind of change? If so @umapraseeda could you please check this?

**Comment by umapraseeda** [comment]
> Not sure about the changelog entry, @rlubos do we usually track this kind of change? If so @umapraseeda could you please check this?
Discussed this with @rlubos. We agree that this doesn't have to be mentioned in the changelog, as it is not a feature change and it does not affect the users.

**Comment by richardmccrae-nordicsemi** [comment]
Sounds good @umapraseeda , i believe it was the compliance test that claimed that a changelog entry was required.  ill remove the edit to the changelog and commit

**Review by lemrey — APPROVED** [review]

**Review by shantha-14 — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**